### PR TITLE
[GPII-3516]: Use auth user email for Stackdriver notifications

### PR DESCRIPTION
**Pull request by natarajaya**
This switches Stackdriver's email notification channel to use auth user email for dev clusters.
Unfortunately Terragrunt interpolations [are not supported](https://github.com/gruntwork-io/terragrunt#interpolation-syntax) in module configuration block, so I had to add condition on `env` directly into `gcp-stackdriver-monitoring` module.

I also used `alerts+ENV@raisingthefloor.org` instead of `alerts+DOMAIN_NAME@raisingthefloor.org` to set notification email for all environments on Terragrunt level.

### CODE REVIEW

**Comment by stepanstipl** [comment]
boo `Terragrunt`! 

LGTM otherwise.

**Comment by amatas** [comment]
LGTM

File: gcp/modules/gcp-stackdriver-lbm/main.tf [hunk 1]
```diff
@@ -7,6 +7,7 @@ variable "project_id" {}
 variable "serviceaccount_key" {}
 
 # Enables debug mode when TF_VAR_stackdriver_debug is not empty
+
```
Comment (mrtyler) [inline gcp/modules/gcp-stackdriver-lbm/main.tf]: Extra newline?

File: gcp/live/dev/k8s/stackdriver/monitoring/terraform.tfvars [hunk 2]
```diff
@@ -20,3 +20,4 @@ terragrunt = {
 # ↓ Module configuration (empty means all default)
 
 ssl_enabled_uptime_checks = false
+notification_email = "alerts+dev@raisingthefloor.org"
```
Comment (mrtyler) [inline gcp/live/dev/k8s/stackdriver/monitoring/terraform.tfvars]: If I'm reading this correctly, this value should never be used. If this is correct, can you make it clearer that if we are seeing this value in an alert email then something has gone wrong? `alerts+dev-this-value-should-have-been-overridden@` or something?

File: gcp/modules/gcp-stackdriver-monitoring/main.tf [hunk 3]
```diff
@@ -23,6 +28,7 @@ resource "template_dir" "resources" {
     project_id                = "${var.project_id}"
     domain_name               = "${var.domain_name}"
     ssl_enabled_uptime_checks = "${var.ssl_enabled_uptime_checks}"
+    notification_email        = "${var.env == "dev" ? var.auth_user_email : var.notification_email}"
```
Comment (mrtyler) [inline gcp/modules/gcp-stackdriver-monitoring/main.tf]: How about changing this to `notification_email = "${var.notification_email != ""} ? var.notification_email : var.auth_user_email"`?

File: gcp/modules/gcp-stackdriver-monitoring/main.tf [hunk 4]
```diff
@@ -2,15 +2,20 @@ terraform {
...
+variable "auth_user_email" {}
 
-# Terragrunt variable
+# Terragrunt variables
+
+variable "notification_email" {}
 variable "ssl_enabled_uptime_checks" {}
 
 # Enables debug mode when TF_VAR_stackdriver_debug is not empty
+
```
Comment (mrtyler) [inline gcp/modules/gcp-stackdriver-monitoring/main.tf]: Extra newline?

File: gcp/modules/gcp-stackdriver-lbm/main.tf (hunk 1 above)
Comment (natarajaya) [reply gcp/modules/gcp-stackdriver-lbm/main.tf]: Terraform freaks out and complains about broken formatting without this newline.

File: gcp/live/dev/k8s/stackdriver/monitoring/terraform.tfvars (hunk 2 above)
Comment (natarajaya) [reply gcp/live/dev/k8s/stackdriver/monitoring/terraform.tfvars]: Good idea.

File: gcp/modules/gcp-stackdriver-monitoring/main.tf (hunk 4 above)
Comment (natarajaya) [reply gcp/modules/gcp-stackdriver-monitoring/main.tf]: Added to [match the style](https://github.com/gpii-ops/gpii-infra/pull/241#discussion_r242371032).

File: gcp/modules/gcp-stackdriver-monitoring/main.tf (hunk 3 above)
Comment (natarajaya) [reply gcp/modules/gcp-stackdriver-monitoring/main.tf]: This looks much better, since we don't have to rely on env. Thanks!

**Pull request merged**